src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -474,6 +474,66 @@ protected override TextReader CreateTextReaderImplementation()
             return reader;
         }
 
+        /// <summary>
+        /// Asynchronously creates <see cref="Stream"/> for reading an inline stream property.
+        /// </summary>
+        /// <returns>
+        /// A task that represents the asynchronous operation.
+        /// The value of the TResult parameter contains a <see cref="Stream"/> for reading the stream property.
+        /// </returns>
+        protected override async Task<Stream> CreateReadStreamImplementationAsync()
+        {
+            Stream stream;
+            IJsonStreamReaderAsync streamReader = this.jsonLightInputContext.JsonReader as IJsonStreamReaderAsync;
+            if (streamReader != null)
+            {
+                stream = await streamReader.CreateReadStreamAsync()
+                    .ConfigureAwait(false);
+            }
+            else
+            {
+                // JsonReader doesn't support streaming; read as a string and convert
+                // Skip over property or start array
+                await this.jsonLightInputContext.JsonReader.ReadAsync()

```

Comment:
@habbes If you can recall our session on OData JSON writers, you can substitute the inbuilt `JsonReader` by injecting your own custom reader. The custom reader may not necessarily inherit from `IJsonStreamReader` and `IJsonStreamReaderAsync`. It wouldn't be a simple effort but I can actually implement a custom reader and use that to test this code block. In any case I was planning to do that when making changes to `ODataJsonLightInputContext`.

Reply:
@habbes Like we observed, this could easily be dead code. We'll revisit this at a later date to either remove the block altogether or to make changes to make the code reachable